cmd/kube-controller-manager/app/core.go

```diff
@@ -143,14 +151,42 @@ func startNodeIpamController(ctx ControllerContext) (http.Handler, bool, error)
 		}
 	}
 
+	nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6 := defaultNodeMaskCIDRIPv4, defaultNodeMaskCIDRIPv6
+
+	// NodeCIDRMaskSize can be used only for single stack clusters
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) {
+		if ctx.ComponentConfig.NodeIPAMController.NodeCIDRMaskSize != 0 {
```

Comment:
can we factor these out into their own funcs?
```
func getNodeCIDRMaskSizes(cfg ..., ipv4Mask *int, ipv6Mask *int) error ...
func getNodeCIDRMaskSizesDualStack(cfg ..., ipv4Mask *int, ipv6Mask *int) error ...
```

Reply:
thanks @bowei! done.